src/run.js

```diff
@@ -182,10 +183,14 @@ const processPage = ({
       // To build up a map of all downloaded CSS
       page.on('response', response => {
         const responseUrl = response.url()
+
         const ct = response.headers()['content-type'] || ''
-        if (!response.ok()) {
+        if (response.status() >= 400) {
```

Comment:
thing is that even html redirects go through this logic. You can simply move this block below `if (ct.indexOf('text/css') > -1 || /\.css$/i.test(responseUrl)) {` and it will work for html redirect case without additional logic. But it doesn't work for css redirects case

Reply:
I'm not entirely sure I understood your comment but I poked around and noticed that the problem was that the `Location` header had a relative URL. The lines below, quite nicely, turns it into an absolute URL. That solved the breaking tests.

Reply:
```diff
-        if (!response.ok()) {
-          return safeReject(new Error(`${response.status()} on ${responseUrl}`))
-        }
         if (ct.indexOf('text/css') > -1 || /\.css$/i.test(responseUrl)) {
+          if (!response.ok()) {
+            return safeReject(new Error(`${response.status()} on ${pageUrl}`))
+          }
```

https://github.com/peterbe/minimalcss/pull/107/files#diff-f712f2169bd3dca93334576e35acff9fL186

Reply:
move all logic about `response.status()` below `if (ct.indexOf('text/css') > -1 || /\.css$/i.test(responseUrl)) {`. This logic should be applied only for CSS documents, nothing else

Reply:
I think I understand what you mean. However, it sounds risky. 
What if I have: `<link rel="stylesheet" href="http://example.com/redirect-to-the-real-css.cgi">`. 
When this resource is coming through our `page.on('response'` handler, its filename doesn't end with `.css` and it's content type is not going to be `text/css`.

For example:
```sh
▶ curl -v http://cdn-2916.kxcdn.com/static/css/base.min.7d48dc16ca51.css\?cachebust\=1 > /dev/null
[snip]
> GET /static/css/base.min.7d48dc16ca51.css?cachebust=1 HTTP/1.1
> Host: cdn-2916.kxcdn.com
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 301 Moved Permanently
< Server: keycdn-engine
< Date: Fri, 26 Jan 2018 15:24:10 GMT
< Content-Type: text/html
< Content-Length: 178
< Connection: keep-alive
< Location: https://cdn-2916.kxcdn.com/static/css/base.min.7d48dc16ca51.css?cachebust=1
< X-Edge-Location: usch
< Access-Control-Allow-Origin: *
<
```
If I only wrote down where this redirect **if** it's `.css` or `text/css` content-type, then this would fail. 

By writing down every single redirect, we're safer.

Reply:
We can add test for this case

Reply:
Tested for in the latest push.